Wykonano zadanie

a-dreszler opened the pull request.

a-dreszler committed:
Wykonano zadanie

slawekludwiczak reviewed (changes requested):
Klasa Memory nie powinna dziedziczyć po Processor. Wprowadź dodatkową klasę OverclockableComponent i postaraj się zaimplementować wzorzec metody szablonowej.

Klasę Main przenieś do innego pakietu

https://javastartbootcamp.slack.com/archives/C04LP8YSVQX/p1682408292077469

a-dreszler committed:
Wprowadzono poprawki:
 > Klasy processor i memory dziedziczą teraz po klasie abstrakcyjnej OverclockableComponent
 > Do klasy OverclockableComponent przeniesiono wszystkie funkcjonalności związane z podkręcaniem
 > Zwiększenia taktowania i temperatury są teraz pobierane z metod szablonowych getTempIncrease i getOverclockIncrement które mają oddzielne implementacje w klasach Memory i Processor
 > Settery pól clockRate i temperature są od teraz prywatne
